blocks/serialization.py

```diff
@@ -71,17 +169,73 @@ def secure_pickle_dump(object_, path):
         The object to be saved to the disk.
     path : str
         The destination path.
+    dump_function : function
+        The function that is used to perform the serialization. Must take
+        an object and file object as arguments. By default, :func:`dump` is
+        used. An alternative would be :func:`pickle.dump`.
 
     """
     try:
-        # Use the same destination directory, as /tmp can be too
-        # small.  This also make the move to copy if the destination
-        # wasn't on the same partition.
-        with tempfile.NamedTemporaryFile(delete=False,
-                                         dir=os.path.dirname(path)) as temp:
-            pickle_dump(object_, temp)
+        with tempfile.NamedTemporaryFile(delete=False) as temp:
+            dump_function(object_, temp)
         shutil.move(temp.name, path)
     except:
         if "temp" in locals():
             os.remove(temp.name)
         raise
+
+
+def continue_training(path):
+    """Continues training using checkpoint.
+
+    Parameters
+    ----------
+    path : str
+        Path to checkpoint.
+
+    Notes
+    -----
+    Python picklers can unpickle objects from global namespace only if
+    they are present in namespace where unpickling happens. Often global
+    functions are needed for mapping, filtering and other data stream
+    operations. In a case if the main loop uses global objects and
+    this function fails with a message like
+    ```
+    AttributeError: 'module' object has no attribute '...'
+    ```
+    it means that you need to import these objects.
+
+    Examples
+    --------
+    This function can be used in two ways: in your script where a main
+    loop defined or in a different script. For later options see Notes
+    section.
+
+    """
+    with change_recursion_limit(config.recursion_limit):
+        with open(path, "rb") as f:
+            main_loop = load(f)
+    main_loop.run()
+
+
+def load_parameter_values(path):
+    """Load parameter values saved by :func:`dump`.
+
+    This is a thin wrapper over :func:`numpy.load`. It changes the names of
+    the arrays to ones compatible with :meth:`.Model.set_param_values`.
+
+    Parameters
+    ----------
+    path : str or file
+        The source for loading from.
+
+    Returns
+    -------
+    A dictionary of (parameter name, numpy array) pairs.
+
+    """
+    source = numpy.load(path)
+    param_values = {'/' + name.replace(BRICK_DELIMITER, '/'): value
+                    for name, value in source.items() if name != 'pkl'}
+    source.close()
```

Comment:
From my point of view, `with` statement is cleaner and more robust for typos in future changes.